packages/common/src/lib/common/interview.ts

```diff
@@ -0,0 +1,74 @@
+import rl from 'readline-sync';
+import fs from 'fs-extra';
+
+/** Regex to limit inputs to yes/no shortcuts */
+const YES_NO_REGEX = /^[YyNn]?$/;
+
+type SharedDatabaseOptions = ioBroker.ObjectsDatabaseOptions | ioBroker.StatesDatabaseOptions;
+
+interface DefaultRedisInterviewOptions<TConfig extends SharedDatabaseOptions> {
+    /** Database type */
+    type: 'objects' | 'states';
+    /** Prefilled config */
+    config: TConfig;
+}
+
+/**
+ * The default redis interview shared between objects and states
+ *
+ * @param options config and type information
+ */
+export async function defaultRedisInterview<TConfig extends SharedDatabaseOptions>(
+    options: DefaultRedisInterviewOptions<TConfig>
+): Promise<TConfig> {
+    const { type, config } = options;
+
+    let answer = rl.question(`Dou you use a TLS connection for your "${type}" redis-server? [y/N]:`, {
+        limit: YES_NO_REGEX,
+        defaultInput: 'N'
+    });
+
+    if (answer.toLowerCase() === 'n') {
+        return config;
+    }
+
+    config.options.tls = {};
+
+    answer = rl.question(`Dou you use a self-signed certificate for your "${type}" redis-server? [y/N]:`, {
+        limit: YES_NO_REGEX,
+        defaultInput: 'N'
+    });
+
+    if (answer.toLowerCase() === 'y') {
+        config.options.tls.rejectUnauthorized = false;
+    }
+
+    answer = rl.question(`Please specify the path to your "${type}" redis-server "certificate" file:`);
+
+    try {
+        const certContent = await fs.readFile(answer, { encoding: 'utf8' });
+        config.options.tls.cert = certContent;
+    } catch (e) {
+        console.warn(`Could not read the "certificate" file, cert will be left empty: ${e.message}`);
+    }
+
+    answer = rl.question(`Please specify the path to your "${type}" redis-server "key" file:`);
+
+    try {
+        const keyContent = await fs.readFile(answer, { encoding: 'utf8' });
+        config.options.tls.key = keyContent;
+    } catch (e) {
+        console.warn(`Could not read the "key" file, cert will be left empty: ${e.message}`);
+    }
+
+    answer = rl.question(`Please specify the path to your "${type}" redis-server "CA" file:`);
+
+    try {
+        const caContent = await fs.readFile(answer, { encoding: 'utf8' });
+        config.options.tls.ca = caContent;
+    } catch (e) {
+        console.warn(`Could not read the "CA" file, cert will be left empty: ${e.message}`);
```

Comment:
same here

Reply:
here it is already last question so doesn't matter